Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Rating.hs

```diff
@@ -70,24 +71,30 @@ handler merchantId req ride = do
       QRating.updateRating ratingValue feedbackDetails isSafe issueId wasOfferedAssistance rideRating.id driverId
       logDebug "Driver Rating Coin Event"
       fork "DriverCoinRating Event" $ DC.driverCoinsEvent driverId merchantId ride.merchantOperatingCityId (DCT.Rating ratingValue ride.chargeableDistance)
-  calculateAverageRating driverId merchant.minimumDriverRatesCount
+  calculateAverageRating driverId merchant.minimumDriverRatesCount ratingValue person.totalRatings person.totalRatingScore
 
 calculateAverageRating ::
   (CacheFlow m r, EsqDBFlow m r, EncFlow m r) =>
   Id DP.Person ->
   Int ->
+  Int ->
+  Int ->
+  Int ->
   m ()
-calculateAverageRating personId minimumDriverRatesCount = do
+calculateAverageRating personId minimumDriverRatesCount ratingValue totalRatings totalRatingScore = do
   logTagInfo "PersonAPI" $ "Recalculating average rating for driver " +|| personId ||+ ""
-  allRatings <- B.runInReplica $ QRating.findAllRatingsForPerson personId
-  let ratingsSum = fromIntegral $ sum (allRatings <&> (.ratingValue))
-  let ratingCount = length allRatings
-  when (ratingCount == 0) $
+  let newRatingsCount = totalRatings + 1
+  let newTotalRatingScore = totalRatingScore + ratingValue
+  when (totalRatings == 0) $
     logTagInfo "PersonAPI" "No rating found to calculate"
-  when (ratingCount >= minimumDriverRatesCount) $ do
-    let newAverage = ratingsSum / fromIntegral ratingCount
-    logTagInfo "PersonAPI" $ "New average rating for person " +|| personId ||+ " , rating is " +|| newAverage ||+ ""
-    void $ QP.updateAverageRating personId newAverage
+  if totalRatings >= minimumDriverRatesCount
```

Comment:
@vsaimanohar It can be simplified by returning isValidRating from if statement, as rest of the part is same.